src/server/scripts/Outland/zone_nagrand.cpp

```diff
@@ -636,6 +636,40 @@ class go_warmaul_prison : public GameObjectScript
         }
 };
 
+// 32307 - Plant Warmaul Ogre Banner
+class spell_q9927_plant_warmaul_ogre_banner : public SpellScript
+{
+    PrepareSpellScript(spell_q9927_plant_warmaul_ogre_banner);
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        if (Player* caster = GetCaster()->ToPlayer())
+            caster->RewardPlayerAndGroupAtEvent(18388, caster);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_q9927_plant_warmaul_ogre_banner::HandleHit, EFFECT_1, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+// 32314 - Plant Kil'Sorrow Banner
+class spell_q9931_plant_kilsorrow_banner : public SpellScript
+{
+    PrepareSpellScript(spell_q9931_plant_kilsorrow_banner);
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        if (Player* caster = GetCaster()->ToPlayer())
+            caster->RewardPlayerAndGroupAtEvent(18393, caster);
```

Comment:
I think you should name these 2 numbers

```cpp
 enum PlantBanners
 {
     NPC_XYZ                     = ...,
     QUEST_XYZ                   = ...,
};
```

Reply:
```cpp
enum PlantBannerCredits
{
    NPC_SHADOW_COUNCIL_CREDIT_MARKER = 18388,
    NPC_WARMAUL_OGRE_CREDIT_MARKER   = 18393
};

Reply:
Thanks, was too lazy and I thought he could figure out by himself 🎃

Reply:
Maybe busy with more "important" stuff than looking into these details.

Reply:
Busy with being sick and work, but have some time off now.